charts/redpanda/templates/statefulset.yaml

```
@@ -283,7 +283,7 @@ spec:
             - name: {{ lower $name }}
               containerPort: {{ $listener.port }}
   {{- range $externalName, $external := $listener.external }}
-    {{- if $external.port }}
+  {{- if or ($external.enabled) (and ($values.external.enabled) (not (eq $external.enabled false))) }}
```

Thank you for the fix!

My only concern is that previous check is not longer there.

The following helm values would generate invalid configuration:

```yaml
listeners:
  admin:
    external:
      test:
        enabled: true
```

Can the check be combined?
```suggestion
  {{- if and $external.port (or $external.enabled (and $values.external.enabled (dig "enabled" true $external))) }}
```

P.S. 
I removed the parentheses.
I replaced `not (eq $external.enabled false)` with `dig` as it's easer to understand.

REF:
https://helm.sh/docs/chart_template_guide/function_list/#dig